docs/command_line_reference.rst

```diff
@@ -509,3 +510,67 @@ When you run ``esrally list races``, this will show up again::
     20160518T112341Z  pmc                         append-no-conflicts  defaults  disk:SSD,data_node_count:4
 
 This will help you recognize a specific race when running ``esrally compare``.
+
+.. _command_line_reference_advanced_topics:
+
+Advanced topics
+---------------
+
+``target-hosts``
+~~~~~~~~~~~~~~~~
+
+Rally can also create client connections for multiple Elasticsearch clusters.
+This is only useful if you want to create :ref:`custom runners <adding_tracks_custom_runners>` that execute operations against remote clusters, for example for Cross Cluster Search or Cross Cluster Replication.
+
+To define the host:port pairs for additional clusters with ``target-hosts`` you can specify either a JSON filename (ending in ``.json``) or an inline JSON string. The JSON object should be a collection of name:value pairs. ``name`` is string for the cluster name and there **must be** one ``default``.
+
+Examples:
+
+* json file: ``--target-hosts="target_hosts1.json"``::
+
+    { "default": ["127.0.0.1:9200","10.127.0.3:19200"] }
+
+* json file defining two clusters: ``--target-hosts="target_hosts2.json"``::
+
+    {
+      "default": [
+        {"host": "127.0.0.1", "port": 9200},
+        {"host": "127.0.0.1", "port": 19200}
+      ],
+      "remote":[
+        {"host": "10.127.0.3", "port": 9200},
+        {"host": "10.127.0.8", "port": 9201}
+      ]
+    }
+
+* json inline string defining two clusters::
+
+    --target-hosts="{\"default\":[\"127.0.0.1:9200\"],\"remote\":[\"127.0.0.1:19200\",\"127.0.0.1:19201\"]}"
+
+.. NOTE::
+   **All** internal :ref:`track operations <track_operations>` will use the connection to the ``default`` cluster. However, you can utilize the client connections to the additional clusters in your :ref:`custom runners <adding_tracks_custom_runners>`.
+
+``client-options``
+~~~~~~~~~~~~~~~~~~
+
+``client-options`` can optionally specify options for the Elasticsearch clients when multiple clusters have been defined with ``target-hosts``. If omitted, the default is ``timeout:60`` for all cluster connections.
+
+The format is similar to ``target-hosts``, supporting both filenames ending in ``.json`` or inline JSON, however, the parameters are a collection of name:value pairs, as opposed to arrays.
+
+Examples, assuming that two clusters have been specified with ``--target-hosts``:
+
+* json file: ``--client-options="client_options1.json"``::
+
+    {
+      "default":
+        {"timeout": 60},
+      "remote":
+        {"use_ssl":true,"verify_certs": false,"ca_certs":"/path/to/cacert.pem"}
+    }
+
+* json inline string defining two clusters::
+
+    --client-options="{\"default\":{\"timeout\": 60}, \"remote\": {\"use_ssl\":true,\"verify_certs\":false,\"ca_certs\":\"/path/to/cacert.pem\"}}"
+
+.. WARNING::
+   If you use ``client-options`` you must specify options for **every** cluster name defined with ``target-hosts``. An error message will be displayed if there is a mismatch.
```

Comment:
Nit: Use active voice? "An error message will be displayed if there is a mismatch." -> (something along the lines of): "Rally raises an error on mismatch."?

Reply:
After checking the code I am also not sure where it would actually raise an error on mismatch (I spotted this because I think such a test is missing).

Reply:
The test is here: https://github.com/dliappis/rally/blob/bdc3f0e17632f5a82ae900acf9b8e142fe573934/esrally/rally.py#L690-L691